src/features/game/lib/gameMachine.ts

```diff
@@ -219,6 +224,41 @@ export function startGame(authContext: Options) {
             },
           },
         },
+        loaded: {
+          always: [
+            {
+              target: "announcing",
+              cond: () => {
+                const lastRead = localStorage.getItem("announcementLastRead");
+
+                if (lastRead) {
+                  return (
+                    new Date(lastRead) <
+                    announcements[announcements.length - 1].date
+                  );
+                }
+                return true;
+              },
+            },
+            {
+              target: "playing",
+            },
+          ],
+        },
+        announcing: {
+          on: {
+            ACKNOWLEDGE: {
+              actions: [
+                () =>
+                  localStorage.setItem(
```

Comment:
Consider abstracting data storage operations to a separate file. This will help keep the state machine focussed on states and transitions.

```
// acknowledge is an external file
actions: acknowledge,
```